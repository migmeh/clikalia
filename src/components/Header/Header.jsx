import React from 'react';
import './Header.scss';

export default class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            searchPokemon: "",
        }
        this.handleChange = this.handleChange.bind(this);
        this.onSearch = this.onSearch.bind(this);
    }

    handleChange(e) {
        e.preventDefault();
        this.setState({searchPokemon: e.target.value});
    }

    keyPress(e){
        if(e.keyCode === 13){
            this.onSearch()
        }
    }

    onSearch() {
        this.props.showModal(this.state.searchPokemon.toLowerCase())
    }

    render () {
        return (
            <div className="header">


                <div className="search">
                    <input
                        type="text"
                        className="search-filter"
                        placeholder="PokÃ©mon..."
                        onChange={this.handleChange.bind(this)}
                        onKeyDown={this.keyPress.bind(this)}
                    />
                    <button type="submit" className="search-button" onClick={() => this.onSearch()}>
                        <i className="fa fa-search search-icon"></i>
                    </button>
                </div>

            </div>
        );
    }
}
