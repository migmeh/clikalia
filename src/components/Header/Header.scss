@import '../../assets/sass/globals.scss';

.header {
    background-color: $dark-gray;
    color: white;
    width: 100%;

    z-index:150;
    justify-content: space-between;

    .logo-content{
        display: flex;
    }
    .title{
        font-size: 20px;
        padding: 18px 10px 0 10px;
    }



    .search {
        float:right;
        margin: 0 2rem;
    }

    .search-filter {
        border: 2px solid #566CD6;
        border-right: none;
        padding: 5px;
        border-radius: 5px 0 0 5px;
        outline: none;
        color: $dark-gray;
        margin: 14px 0;
    }

    .search-filter:focus{
        color: $dark-gray;
    }

    .search-icon {
        width: 45px;
        height: 38px;
        border: 1px solid #566CD6;
        background: #566CD6;
        text-align: center;
        color: #fff;
        border-radius: 0 50% 50% 0;
        font-size: 20px;
        vertical-align: middle;
    }

    .search-button {
        background: none;
        color: inherit;
        border: none;
        padding: 0;
        font: inherit;
        outline: inherit;
        cursor: pointer;
        height: 10px;
        margin: 14px 0;
    }

    .fa {
        padding-top: 0.4rem;
    }

}
