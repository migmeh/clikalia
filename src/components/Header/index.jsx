import { connect } from 'react-redux'
import { 
    showModal, 
} from '../../actions/Modal'
import Header from './Header'

const mapStateToProps = (state) => {
    return {
        show_modal: state.modal.show_modal,
    }
}

const mapDispatchToProps = (dispatch) => {
  return {
    showModal: (pokemon_name) => {
        dispatch(showModal(pokemon_name));
    },
  }
}


export default connect(mapStateToProps, mapDispatchToProps)(Header);