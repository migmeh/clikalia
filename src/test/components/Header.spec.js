import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../components/Header/Header';

describe('Header', () => {
  it('Should render correctly', () => {
    const tree = shallow(<Header />);
    expect(tree).toMatchSnapshot();
  });
});