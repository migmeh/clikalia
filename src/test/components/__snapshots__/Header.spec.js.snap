// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render correctly 1`] = `
<div
  className="header"
>
  <div
    className="search"
  >
    <input
      className="search-filter"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="PokÃ©mon..."
      type="text"
    />
    <button
      className="search-button"
      onClick={[Function]}
      type="submit"
    >
      <i
        className="fa fa-search search-icon"
      />
    </button>
  </div>
</div>
`;
